File: ETourBackend/src/main/java/com/example/demo/entity/Journey.java
package com.tourism.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "journey")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Journey {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "journey_id")
    private Integer journeyId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "tour_id")
    private Tour tour;

    @Column(name = "source")
    private String source;

    @Column(name = "destination")
    private String destination;

    @Column(name = "transport")
    private String transport;

    @Lob
    @Column(name = "details")
    private String details;
}

File: ETour-DotNet/Models/Journey.cs
namespace ETour.Api.Models;

public class Journey
{
    public int JourneyId { get; set; }
    public int? TourId { get; set; }
    public Tour Tour { get; set; }
    public string Source { get; set; }
    public string Destination { get; set; }
    public string Transport { get; set; }
    public string Details { get; set; }
}
